### ISSUE TITLE
Official terraform 0.13.2 for OpenBSD is broken 

### ISSUE THREAD

**Issue by pannon** [hashicorp/terraform#26175]
The official OpenBSD binary https://releases.hashicorp.com/terraform/0.13.2/terraform_0.13.2_openbsd_amd64.zip is crashing 100% of the time and is unusable.

### Terraform Version

v0.13.2

### Crash Output

```
panic: cannot determine user directories on OS "openbsd"

goroutine 1 [running]:
github.com/apparentlymart/go-userdirs/userdirs.forApp(...)
	/home/circleci/project/project/vendor/github.com/apparentlymart/go-userdirs/userdirs/app_stub.go:19
github.com/apparentlymart/go-userdirs/userdirs.ForApp(...)
	/home/circleci/project/project/vendor/github.com/apparentlymart/go-userdirs/userdirs/app.go:23
main.implicitProviderSource(0xc00092cf00, 0xc00063fbb8, 0x40cc08)
	/home/circleci/project/project/provider_source.go:158 +0x204
main.providerSource(0x0, 0x0, 0x0, 0xc00092cf00, 0x0, 0x40, 0x203000, 0xc00063fca0, 0x40c396)
	/home/circleci/project/project/provider_source.go:28 +0x39
main.wrappedMain(0x0)
	/home/circleci/project/project/main.go:177 +0x477
main.realMain(0x0)
	/home/circleci/project/project/main.go:106 +0xde
main.main()
	/home/circleci/project/project/main.go:42 +0x3a



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Steps to Reproduce
Execute terraform.
```
terraform
```

### References

There is a fix in this PR (just needs merging):
https://github.com/apparentlymart/go-userdirs/pull/1

**Comment by mildwonkey** [comment]
Thank you for reporting this, @pannon, and very sorry for the crash !

I'll flag the PR in the upstream library for review so we can get this looked at.

**Comment by mildwonkey** [comment]
This is fixed in master (v0.14), and I'm also backporting the fix for the next v0.13 release. Thanks again!
